File: ruby/array123.rb
def array123(arr)
	return arr.include?(1) && arr.include?(2) && arr.include?(3)
end
puts array123([1, 1, 2, 3, 1]) ? "pass" : "fail"
puts array123([1, 1, 2, 4, 1]) ? "fail" : "pass"
puts array123([1, 1, 2, 1, 2, 3]) ? "pass" : "fail"

File: php/array123.php
<?php
	function array123($arr)
	{
		return (in_array(1, $arr) && in_array(2, $arr) && in_array(3, $arr));
	}
	echo array123(array(1, 1, 2, 3, 1)) ? "pass" : "fail"; echo "\n";
	echo array123(array(1, 1, 2, 4, 1)) ? "fail" : "pass"; echo "\n";
	echo array123(array(1, 1, 2, 1, 2, 3)) ? "pass" : "fail"; echo "\n";
?>